main/src/ca/uwaterloo/flix/api/lsp/provider/HighlightProvider.scala

```diff
@@ -158,9 +158,15 @@ object HighlightProvider {
   }
 
   private def highlightCase(uri: String, sym: Symbol.CaseSym)(implicit index: Index, root: Root): JObject = {
-    val write = (root.enums(sym.enumSym).cases(sym).loc, DocumentHighlightKind.Write)
+    val write = root.enums.get(sym.enumSym) match {
```

Comment:
feels like a comprehension might be cleaner

Reply:
Yeah, but I think all of this code has to be refactored.